flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobServer.java

```diff
@@ -196,23 +199,28 @@ public void run() {
 	 */
 	@Override
 	public void shutdown() throws IOException {
-
-		this.shutdownRequested = true;
-		try {
-			this.serverSocket.close();
-		} catch (IOException ioe) {
+		if (shutdownRequested.compareAndSet(false, true)) {
+			try {
+				this.serverSocket.close();
+			}
+			catch (IOException ioe) {
 				LOG.debug("Error while closing the server socket.", ioe);
-		}
-		try {
-			join();
-		} catch (InterruptedException ie) {
-			LOG.debug("Error while waiting for this thread to die.", ie);
-		}
+			}
+			try {
+				join();
```

Comment:
It's from the old code and I am not sure if it really needs to stay, but it ensures that the BlobServer thread really finishes when calling the shutdown method (BlobServer is a Thread and because the join is called from outside of the run method it waits for the BlobServer thread to finish).